plugin/forward/connect.go

```diff
@@ -78,12 +78,17 @@ func (p *Proxy) updateRtt(newRtt time.Duration) {
 }
 
 // Connect selects an upstream, sends the request and waits for a response.
-func (p *Proxy) Connect(ctx context.Context, state request.Request, forceTCP, metric bool) (*dns.Msg, error) {
+func (p *Proxy) Connect(ctx context.Context, state request.Request, pf protoFlags, metric bool) (*dns.Msg, error) {
```

Comment:
protoFlags -> options? (could be used for other fluff - should prolly fold in the metrics thing as well?)

Reply:
so far, we always pass true for metric parameter. Are we going to make this parameter configurable in future? Otherwise we can remove it at all